pkg/bootstrapper/bootstrapper.go

```diff
@@ -271,50 +263,121 @@ func (wmcb *winNodeBootstrapper) translateFile(ignitionSource string, fileTransl
+	envVars := make(map[string]string)
+	lines := strings.Split(contents, "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, "Environment=") {
+			// The format of each environment variable line will look like this:
+			// Environment=\"KUBELET_LOG_LEVEL=3\"
+			split := strings.Split(line, "\"")
+			if len(split) != 3 {
+				// continue on error
+				continue
+			}
+			keyValSplit := strings.Split(split[1], "=")
+			if len(keyValSplit) != 2 {
+				continue
+			}
+			envVars[keyValSplit[0]] = keyValSplit[1]
 		}
+	}
+	return envVars
+}
 
-		results := cloudProviderRegex.FindStringSubmatch(unit.Contents)
-		if len(results) == 2 {
-			wmcb.kubeletArgs["cloud-provider"] = results[1]
-		}
+// translateKubeletArg takes a line from a systemd unit file and returns the value of the argument arg
+func translateKubeletArg(arg, line string, envVars map[string]string) (string, error) {
+	// Line can contain an escaped newline such as
+	// "--cloud-provider=aws \\"
+	trimmed := strings.TrimSuffix(line, "\\")
+	keyValSplit := strings.SplitN(trimmed, "--"+arg+"=", 2)
+	if len(keyValSplit) != 2 {
+		return "", fmt.Errorf("line %s did not contain exactly one instance of the expected argument", line)
+	}
+	if !strings.Contains(keyValSplit[1], "${") {
+		return strings.TrimSpace(keyValSplit[1]), nil
+	}
+	// Handle case where value contains an environment variable, such as:
+	// --v=\"${KUBELET_LOG_LEVEL}\"
+	// NOTE: this only handles the case where the environment variable is the full value
+	envVarSplit := strings.Split(keyValSplit[1], "${")
+	if len(envVarSplit) != 2 {
+		return "", fmt.Errorf("more than one environment variable in argument %s", keyValSplit[1])
+	}
+	trimSplit := strings.Split(envVarSplit[1], "}")
+	if len(trimSplit) != 2 {
+		return "", fmt.Errorf("unexpected format %s", envVarSplit[1])
+	}
+	value, ok := envVars[trimSplit[0]]
+	if !ok {
+		return "", fmt.Errorf("value set to environment variable %s, but the variable was not found", trimSplit[0])
+	}
+	return value, nil
+}
 
+// parseSystemdKubeletUnit gathers the kubelet args we need from the kubelet systemd unit file and returns a
+// map of file translations to be added to the existing file translation map
+func (wmcb *winNodeBootstrapper) parseSystemdKubeletUnit(contents string) (map[string]fileTranslation, error) {
+	files := make(map[string]fileTranslation)
+	envVars := extractEnvVarsFromSystemdUnit(contents)
+	for _, line := range strings.Split(contents, "\n") {
 		// Check for the presence of "--cloud-config" option and if it is present append the value to
 		// filesToTranslate. This option is only present for Azure and hence we cannot assume it as a file that
 		// requires translation across clouds.
-		results = cloudConfigRegex.FindStringSubmatch(unit.Contents)
-		if len(results) == 2 {
-			cloudConfFilename := filepath.Base(results[1])
+		if strings.Contains(line, "--"+cloudConfigOption+"=") {
+			val, err := translateKubeletArg(cloudConfigOption, line, envVars)
```

Comment:
naming again, per every line, you could have just parsed it once, if translateKubeletArg is also returning a map.